src/Session/Session.php

```diff
@@ -101,9 +122,154 @@ public function setData($data)
      */
     public function getUser()
     {
-        $user_uuid = $this->get('user_uuid');
-        // TODO: Remove this direct instantiation to make this more testable
-        $user = new User((object)array('id' => $user_uuid));
-        return $user;
+        if ($this->loggedIn()) {
+            $user_id = $this->get('user_id');
+            // TODO: Remove this direct instantiation to make this more testable
+            $user = new User((object)array('id' => $user_id));
+            return $user;
+        }
+    }
+
+    /**
+     * Authorize the current user prior to running a command.  The
+     * Annotated Commands hook manager will call this function during
+     * the pre-validate phase of any command that has an 'authorize'
+     * annotation.
+     *
+     */
+    public function ensureLogin()
```

Comment:
The @hook annotation is on the Authorizer class, which is what the HookManager will call. This makes the comment above potentially a little confusing.  Perhaps it should say that this is called by Authoerizer::ensureLogin(), which has a @hook annotation, meaning the hook manager will call it. Or something like that.

Alternately, you could throw the whole Authorizer class away if you registered this class (Session) with the hook manager.  Then you could just put the @hook annotation on this method.  Currently, the registration of the Authorizer class is kind of a hack; there should be a well-defined way to do this. Maybe put all the hook classes in one place, as is done with the commands. To this end, it's probably a good idea to maintain a thin adapter.